presto-main/src/main/java/com/facebook/presto/execution/SqlTask.java

```diff
@@ -429,10 +436,20 @@ public TaskInfo updateTask(
         return getTaskInfo();
     }
 
-    public ListenableFuture<BufferResult> getTaskResults(OutputBufferId bufferId, long startingSequenceId, DataSize maxSize)
+    public ListenableFuture<BufferResult> getTaskResults(OutputBufferId bufferId, long startingSequenceId, DataSize maxSize, String communicationSlug)
```

Comment:
Probably instead of passing communicationSlug into `getTaskResults`, add a method `validateCommunicationSlug` and call it before we call `getTaskResults`? We need to check slug for other methods as well.